### PR TITLE
chore(clap): updated clap, rust, and added clap derive minimal

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
Name | Overhead (release) | Build (debug) | Parse (release) | Invalid UTF-8 | Downloads | Version
-----|--------------------|---------------|-----------------|---------------|-----------|--------
null | 0 KiB | 173ms *(full)* <br/>107ms *(incremental)* | 5ms | Y | - | -
argh | 37 KiB | 2s *(full)* <br/>125ms *(incremental)* | 5ms | N | Download count | v0.1.12
bpaf | 251 KiB | 593ms *(full)* <br/>144ms *(incremental)* | 8ms | Y | Download count | v0.9.12
bpaf_derive | 246 KiB | 3s *(full)* <br/>138ms *(incremental)* | 6ms | Y | Download count | v0.9.12
clap | 594 KiB | 1s *(full)* <br/>227ms *(incremental)* | 7ms | Y | Download count | v4.5.16
clap-minimal | 366 KiB | 1s *(full)* <br/>221ms *(incremental)* | 4ms | Y | Download count | v4.5.16
clap_derive | 624 KiB | 2s *(full)* <br/>242ms *(incremental)* | 6ms | Y | Download count | v4.5.16
clap_derive-minimal | 392 KiB | 2s *(full)* <br/>206ms *(incremental)* | 7ms | Y | Download count | v4.5.16
clap_lex | 34 KiB | 274ms *(full)* <br/>117ms *(incremental)* | 6ms | Y | Download count | v0.6.0
gumdrop | 31 KiB | 2s *(full)* <br/>125ms *(incremental)* | 5ms | N | Download count | v0.8.1
lexopt | 43 KiB | 285ms *(full)* <br/>118ms *(incremental)* | 3ms | Y | Download count | v0.3.0
pico-args | 26 KiB | 252ms *(full)* <br/>115ms *(incremental)* | 5ms | Y | Download count | v0.5.0
xflags | 30 KiB | 495ms *(full)* <br/>115ms *(incremental)* | 5ms | Y | Download count | v0.3.2

*System: Linux 6.9.3-76060903-generic (x86_64), rustc 1.79.0 (129f3b996 2024-06-10) w/ `-j 32`*

### CODE REVIEW

File: rust-toolchain.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[toolchain]
+# pin to a specific version of the Rust toolchain
+# so that performance more consistent consistent
+# clarifies how rust influences parameters
+channel = "1.80.0"
+components = ["rust-analyzer", "tree"]
```
Comment (epage) [inline rust-toolchain.toml]: I'd rather not have a rust-toolchain.toml file to manage

File: examples/clap_derive-app/Cargo.toml [hunk 2]
```diff
@@ -9,4 +9,4 @@ name = "clap_derive-app"
 path = "app.rs"
```
[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply examples/clap_derive-app/Cargo.toml:9]: So all numbers of derive minimal can be algebraically calculated by simple  average of derive and minimal?

[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply examples/clap_derive-app/Cargo.toml:9]: i do not see that is the case

File: rust-toolchain.toml (hunk 1 above)
Comment (dzmitry-lahoda) [reply rust-toolchain.toml]: benchamarking are scientific measurments? if yes they must be reproducible and use same base line. 
using random rust reduces quality of benchmarks.
canonical latest stable is good target to measure and remeasure.

File: examples/clap_derive-app/Cargo.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (epage) [reply examples/clap_derive-app/Cargo.toml:9]: Not that mathematically precise but I feel like that is getting overly pedantic to get to that data.

**Pull request closed without merging**